arch/c4_context.md

```diff
@@ -0,0 +1,49 @@
+# Level 1: Radix System Context
+
+This page implement the first "C" in the [C4Model](https://c4model.com/#coreDiagrams): the (System) **C**ontext.
+
+All production Systems (External and Radix) and Users that are commonly found on Radix networks are introduced on this page.
+
+![uncached image](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/radixdlt/docs/arch/arch/c4_context.puml)
+
+## Shards and Chunks
+
+Described in detail in the Radix [Whitepaper v2](https://papers.radixdlt.com/tempo/v2/), shards enable Radix Networks to scale infinitely.
+This is achieved by splitting up global state (Ledger) to different nodes. Each node can be part of one or more Shards slices - called Chunks.
+
+The minimal Shard space that a nodes can serve is one Chunk. The number of Chunks served depends on the hardware configuration (CPU, MEM, I/O) of
+a specific node. Each node has advanced algorithms for monitoring its own performance and dynamically - to other nodes non-deterministically - scale up and down
+ the number of chunks served.
+
+### Redundancy
+
+Nodes within the same Shard space are redundant to each other, which means that they serve and validate the same transactions.
+
+## Core Nodes
+
+The first class citizens of the Radix Network. Nodes are organised in Shards. A node can serve ("be in") one or more Shard Chunks.
+
+Core Nodes are responsible for:
+
+* Validating all transactions that are relevant for its Chunks.
+* Store all transactions that are relevant for its Chunks.
+* Allow submission of atoms from Light Clients through a PubSubscribe-style API.
```

Comment:
To be precise, the light client API is less pub/sub and more similar to a state syncing type of API. I think it's an important conceptual difference.

Reply:
Ok cool, can you please point to some blog, wiki that describes this type of API? AFAIK these are the only two well-known types of API communication patterns:

- [request-response](https://en.wikipedia.org/wiki/Request%E2%80%93response)
- [publish-subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern)

We actually support both, but I figured the pub-sub better matches the Rx/Observer pattern that is very typical for us.

Reply:
It's not really an API (as it's less application related) and more of a protocol for real-time state synchronization. Here are some other examples: http://offlinefirst.org/sync/